tests/test_lfs.py

```diff
@@ -74,3 +85,197 @@ def test_lfs(tmp_dir: TmpDir, scm: Git, lfs_objects: TmpDir):
+    "rate_limit_header",
+    [
+        lambda: {"Retry-After": "1"},
+        lambda: {"RateLimit-Reset": f"{int(time()) + 1}"},
+        lambda: {"X-RateLimit-Reset": f"{int(time()) + 1}"},
+    ],
+)
+def test_rate_limit_retry(
+    storage: LFSStorage, rate_limit_header: Callable[[], dict[str, str]]
+):
+    client = LFSClient.from_git_url("http://git.example.com/namespace/project.git")
+    lfs_batch_url = f"{client.url}/objects/batch"
+    lfs_object_url = f"http://git-lfs.example.com/{FOO_OID}"
+    recorder = CallbackResultRecorder()
+
+    with aioresponses() as m:
+        m.post(
+            lfs_batch_url,
+            callback=recorder.record(
+                CallbackResult(
+                    status=HTTPStatus.TOO_MANY_REQUESTS,
+                    headers=rate_limit_header(),
+                    reason="Too many requests",
+                )
+            ),
+        )
+        m.post(
+            lfs_batch_url,
+            callback=recorder.record(
+                CallbackResult(
+                    status=HTTPStatus.OK,
+                    headers={"Content-Type": "application/vnd.git-lfs+json"},
+                    payload={
+                        "transfer": "basic",
+                        "objects": [
+                            {
+                                "oid": FOO_OID,
+                                "size": FOO_SIZE,
+                                "authenticated": True,
+                                "actions": {
+                                    "download": {
+                                        "href": lfs_object_url,
+                                    }
+                                },
+                            }
+                        ],
+                        "hash_algo": "sha256",
+                    },
+                )
+            ),
+        )
+        m.get(
+            lfs_object_url,
+            callback=recorder.record(
+                CallbackResult(
+                    status=HTTPStatus.TOO_MANY_REQUESTS,
+                    headers=rate_limit_header(),
+                    reason="Too many requests",
+                )
+            ),
+        )
+        m.get(
+            lfs_object_url,
+            callback=recorder.record(
+                CallbackResult(
+                    status=HTTPStatus.OK,
+                    body="lfs data",
+                )
+            ),
+        )
+
+        client.download(storage, [Pointer(oid=FOO_OID, size=FOO_SIZE)])
+
+        results = recorder[lfs_batch_url]
+        assert [r.status for r in results] == [429, 200]
+
+        results = recorder[lfs_object_url]
+        assert [r.status for r in results] == [429, 200]
+
+
+@pytest.mark.parametrize(
+    "rate_limit_header",
+    [
+        lambda: {"Retry-After": "1"},
+        lambda: {"RateLimit-Reset": f"{int(time()) + 1}"},
+        lambda: {"X-RateLimit-Reset": f"{int(time()) + 1}"},
+    ],
+)
+def test_rate_limit_max_retries_batch(
```

Comment:
it seems this already covered by the test above?

Reply:
Do you mean the parametrization with the different rate limit headers? Or what specifically do you think is already covered? 🤔 

This test ensures that rate limit retries for the batch API don't continue indefinitely.

Reply:
okay, yes, I see the difference 👍